### PR TITLE
tools/istio-clean-iptables: clean up all DNS UDP rules

### PR DESCRIPTION
**Pull request by kailun-qin**
This patch adds the support of cleaning up all DNS UDP rules in iptables tools,
by reusing the same logic on creation. This helps prevent them from being out
of sync. 

To allow multiple flags with the same name to be used across different
subcommands, this patch also moves all viper mutation and binding to `PreRun`
as part of the command execution so that no overwritten will take place.

Fixes https://github.com/istio/istio/issues/30674

Signed-off-by: Kailun Qin <kailun.qin@intel.com>

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
This is great, thanks! A couple comments.

* Do you think we can do this in a more generic way? Or share the code with the creation. My concern is they will fall out of sync (and actually, it already has from a PR changing the iptables code that merged a couple of days ago).
* I was thinking it would be great to have a test that makes sure all combinations of `istio-iptables; istio-clean-iptables` results in empty table, but I am not sure any easy way to do so without spinning up a docker container and running it. any ideas?

**Comment by kailun-qin** [comment]
> This is great, thanks! A couple comments.
> 

Thanks for the comments!

> * Do you think we can do this in a more generic way? Or share the code with the creation. My concern is they will fall out of sync (and actually, it already has from a PR changing the iptables code that merged a couple of days ago).

Totally agree. I'll give it a shot.

> * I was thinking it would be great to have a test that makes sure all combinations of `istio-iptables; istio-clean-iptables` results in empty table, but I am not sure any easy way to do so without spinning up a docker container and running it. any ideas?

I was thinking about associating w/ tests to make sure the cleanup does correctly and completely for any combination. But I have the same feeling that there is no easy way to add w/o spinning up a docker container. Let's keep this open and see if any further suggestion.

**Comment by kailun-qin** [comment]
Updated using the same logic for DNS UDP rule creation and deletion so that they can be kept in sync.
Looking for a more elegant way, but it seems to me it's hard to implement w/o a further refactoring.

File: tools/istio-clean-iptables/pkg/cmd/cleanup.go [hunk 1]
```diff
@@ -26,13 +29,19 @@ func flushAndDeleteChains(ext dep.Dependencies, cmd string, table string, chains
...
 	for _, table := range []string{constants.NAT, constants.MANGLE} {
-		// Remove the old chains
 		ext.RunQuietlyAndIgnore(cmd, "-t", table, "-D", constants.PREROUTING, "-p", constants.TCP, "-j", constants.ISTIOINBOUND)
 	}
 	ext.RunQuietlyAndIgnore(cmd, "-t", constants.NAT, "-D", constants.OUTPUT, "-p", constants.TCP, "-j", constants.ISTIOOUTPUT)
 
+	redirectDNS := cfg.RedirectDNS
+	// Remove the old DNS UDP rules
+	if redirectDNS {
+		common.HandleDnsUdp(common.DeleteOps, builder.NewIptablesBuilder(), ext, cmd, cfg.ProxyUID, cfg.ProxyGID, cfg.DNSServersV4)
```
Comment (howardjohn) [inline tools/istio-clean-iptables/pkg/cmd/cleanup.go:42]: This ends up running `iptables iptables -t nat -D OUTPUT -p udp --dport 53 -m owner --uid-owner 1337 -j RETURN` (note 2x iptables)

Comment (kailun-qin) [reply tools/istio-clean-iptables/pkg/cmd/cleanup.go]: Good catch, thanks a lot! Making careless mistakes while wrapping up the code... :(

File: pilot/pkg/security/authz/model/generator.go [hunk 2]
```diff
@@ -45,8 +44,7 @@ func (destIPGenerator) principal(_, _ string, _ bool) (*rbacpb.Principal, error)
 	return nil, fmt.Errorf("unimplemented")
 }
 
-type destPortGenerator struct {
-}
+type destPortGenerator struct{}
```
Comment (howardjohn) [inline pilot/pkg/security/authz/model/generator.go:47]: prefer to not change all these unrelated files

**Review by howardjohn — APPROVED** [review]
thanks!!

File: pilot/pkg/security/authz/model/generator.go (hunk 2 above)
Comment (kailun-qin) [reply pilot/pkg/security/authz/model/generator.go]: Thanks! Remove all unrelated changes introduced by local format on `make gen`,
